src/ios/WebviewProxy.m

```diff
@@ -45,51 +45,88 @@ - (BOOL) overrideSchemeTask: (id <WKURLSchemeTask>)urlSchemeTask {
         [request setHTTPShouldHandleCookies:YES];
         [request setTimeoutInterval:1800];
         
-        [[[NSURLSession sharedSession] dataTaskWithRequest:request completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
-            if(error && (self.stoppedTasks == nil || ![self.stoppedTasks containsObject:urlSchemeTask])) {
-                @try {
-                    NSLog(@"WebviewProxy error: %@", error);
-                    [urlSchemeTask didFailWithError:error];
-                    return;
-                } @catch (NSException *exception) {
-                    NSLog(@"WebViewProxy send error exception: %@", exception.debugDescription);
+        NSString *validDomain = request.URL.host;
+        const BOOL requestIsSecure = [request.URL.scheme isEqualToString:@"https"];
+
+        NSMutableArray *array = [NSMutableArray array];
+        
+        // https://stackoverflow.com/a/32845148
+        [cookieStore getAllCookies:^(NSArray* cookies) {
+            // case 1: didn't call this completionHandler
+            for (NSHTTPCookie *cookie in cookies) {
+                NSLog(@"Proxy checking cookie %@", cookie.name);
+                // Don't even bother with values containing a `'`
+                if ([cookie.name rangeOfString:@"'"].location != NSNotFound) {
+                    NSLog(@"Skipping %@ because it contains a '", cookie.properties);
+                    continue;
                 }
+                
+                // Is the cookie for current domain?
+                if (![validDomain hasSuffix:cookie.domain]) {
+                    NSLog(@"Skipping %@ (because not %@)", cookie.properties, validDomain);
+                    continue;
+                }
+                
+                // Are we secure only?
+                if (cookie.secure && !requestIsSecure) {
+                    NSLog(@"Skipping %@ (because %@ not secure)", cookie.properties, request.URL.absoluteString);
+                    continue;
+                }
+                
+                NSString *value = [NSString stringWithFormat:@"%@=%@", cookie.name, cookie.value];
+                NSLog(@"Proxy Adding cookie: %@", cookie.name);
+                [array addObject:value];
             }
+            NSString *header = [array componentsJoinedByString:@";"];
+            NSLog(@"Proxy Setting cookie: %@", header);
+            [request setValue:header forHTTPHeaderField:@"Cookie"];
```

Comment:
Sorry for the late review.

So this basically sets cookies to the request before sending them? In theory they should already be set by WebKit automatically because they are in the cookie store used by the session that is doing the request.

From my experience this is not really necessary and strangely I did not encouter the issues discussed in #3. Maybe something different is the cause of this issue, but this needs further investigation.